### PR TITLE
1982 - Added --force-full option to importer pulp-admin

### PR DESCRIPTION
**Pull request by BrnoPCmaniak**
https://pulp.plan.io/issues/1982

pulp.client.commands.repo.sync_publish.SyncPublishCommand
 * Moved generate_override_config from RunPublishRepositoryCommand
   * SyncPublishCommand now accepts override_config_options
   * SyncPublishCommand now have new method generate_override_config

pulp.client.commands.repo.sync_publish.RunSyncRepositoryCommand(pulp-admin * repo sync)
 * Now accepts --force-full flag

### CODE REVIEW

File: client_lib/pulp/client/commands/repo/sync_publish.py [hunk 1]
```diff
@@ -221,6 +275,7 @@ def run(self, **kwargs):
         repo_id = kwargs[options.OPTION_REPO_ID.keyword]
         background = kwargs[polling.FLAG_BACKGROUND.keyword]
         override_config = {}
+        print("self.override_config_keywords" + str(self.override_config_keywords))
```
Comment (goosemania) [inline client_lib/pulp/client/commands/repo/sync_publish.py]: Please remove this `print`

File: docs/dev-guide/integration/rest-api/repo/cud.rst [hunk 2]
```diff
@@ -155,6 +155,12 @@ dictate how the importer will function when it synchronizes content. The possibl
 configuration values are contingent on the type of importer being added; each
 importer type will support a different set of values relevant to how it functions.
 
+.. note::
+        There is an optional importer configuration parameter that is usable
+        on all importer plugin types called ``force_full``. If this parameter is set,
+        every sync of the repo using this importer will be done from scratch.
+        More details are `here <http://pulp.readthedocs.org/en/latest/user-guide/release-notes/2.9.x.html#pulp-2-9-0>`_.
```
Comment (goosemania) [inline docs/dev-guide/integration/rest-api/repo/cud.rst]: I am not sure I understand what is sync "from scratch", you probably meant that sync will be operational. Is my understanding correct?

File: docs/user-guide/release-notes/2.9.x.rst [hunk 3]
```diff
@@ -15,3 +15,9 @@ New Features
   `--force-full` via CLI.
 
 * A new :ref:`security disclosures` process has been put in place.
+
+* Re-sync of the repository will be a no operation by default if since the last sync, no units
+  were added/removed and there were no changes made to the importer settings. A sync which
+  causes no operation has a skipped result and therefore does not re-create repodata files.
```
Comment (goosemania) [inline docs/user-guide/release-notes/2.9.x.rst]: I do not think sync creates any files, this is done during publish.

Comment (dralley) [inline docs/user-guide/release-notes/2.9.x.rst]: Also, this should go in the 2.10.x release notes

Comment (goosemania) [inline docs/user-guide/release-notes/2.9.x.rst]: @dralley, I agree, thanks. And this PR should be against the master branch

**Comment by goosemania** [comment]
Please re-open this PR against master branch

**Pull request closed without merging**